react-time-track/src/views/edit-user.js

```diff
@@ -0,0 +1,67 @@
+/** @jsx jsx */
+import React from "react";
+import {jsx} from "@emotion/core";
+import {navigate} from "@reach/router";
+import {Li, Title, UserCard, Card, Button} from "../components/ui";
+import {getUser, editUser} from "../services/user";
+
+function EditUser({user_id}) {
+    const [user, setUser] = React.useState({
+			name: "",
+			email: "",
+			role: "",
+			rate: 0
+		});
+
+    React.useEffect(()=> {
+			console.log(user_id)
+      getUser(user_id)
+      .then(data => {setUser(data)
+    	});
+		}, []);
+
+	React.useEffect(()=> {
+		console.log(user);
+	}, [user]);
+
+	function handleChange(e, key) {
+		setUser({...user, [key]: e.target.value})
```

Comment:
Hi, I think this is great. Could you explain me the benefits of it?

Reply:
Sure. `handleChange` function use `e` and `key` as parameters. Firstly, I passed the `user` object properties using a `spread operator`, so I can access to its properties. Then, I pass `[key]` & `e.target.value` to hear all the changes in its `keys` and its `values` in order to replace them with the new data.